### PR TITLE
Quest proposal after confirmation fixed

### LINKED ISSUE #714: Quest proposal after confirmation
Capture d’écran 2024-05-31 à 16 18 17

It just proposed me the Bountive quest after finishing the bountive quest :/

### PR DESCRIPTION
**Pull request by mystical-prog**
# Small Bug Fix

This PR solves the suggested quests after completing one, before the fix it was suggesting the same completed quest to be done again but it has been solved now. 
I have made the completed route dynamic now and it now takes the quest id of the completed quest as route parameter and the uncompleted quests function will now exclude the quest id in the router param.

Resolves: #714 

## Summary by CodeRabbit

- **New Features**
	- Enhanced quest completion page to dynamically exclude specific quests based on their ID.
	- Updated quest reward navigation to include the quest ID in the URL path for better tracking and user experience.

### CODE REVIEW

**Comment by vking45** [comment]
@ayushtom can you please look into this PR ?

**Comment by mystical-prog** [comment]
@ayushtom can you please look into this PR

File: .env.example
Comment (ayushtom) [inline .env.example:1]: why delete this?

File: package-lock.json
Comment (ayushtom) [inline package-lock.json:1]: why change this file?

**Review by ayushtom — CHANGES_REQUESTED** [review]

**Comment by mystical-prog** [comment]
there were some errors from my side, making the requested changes

**Comment by mystical-prog** [comment]
@ayushtom can you please look into it now ?

**Pull request closed without merging**